File: src/main/java/com/example/blog/controller/AccountController.java
package com.example.blog.controller;

import com.example.blog.dto.account.*;
import com.example.blog.service.AccountService;

import javax.validation.Valid;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("account")
public class AccountController {
    private final AccountService accountService;

    AccountController(AccountService accountService) {
        this.accountService = accountService;
    }

    @PostMapping("/signup")
    @ResponseBody
    @ResponseStatus(HttpStatus.CREATED)
    public AccountDto signUp(@Valid @RequestBody SignUpRequest req) {
        AccountDto accountDto = accountService.createAccount(req.getAccountName(), req.getPassword(), req.getEmail());
        return accountDto;
    }

    @PostMapping("/signin")
    @ResponseBody
    @ResponseStatus(HttpStatus.OK)
    public AuthDto signIn(@Valid @RequestBody SignInRequest req) {
        AuthDto authDto = accountService.accountValidation(req.getAccountName(), req.getPassword());
        return authDto;
    }

    @GetMapping
    @ResponseBody
    @ResponseStatus(HttpStatus.OK)
    public String AuthCheck() {
        return "OK";
    }
}

File: src/main/kotlin/com/example/blog/controller/AccountController.kt
package com.example.blog.controller

import com.example.blog.dto.AccountDto
import com.example.blog.dto.AuthDto
import com.example.blog.dto.SignInRequest
import com.example.blog.dto.SignUpRequest
import com.example.blog.service.AccountService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

@RestController
@RequestMapping("/api/account")
class AccountController(
    private val accountService: AccountService
) {
    @PostMapping("/signup")
    @ResponseStatus(HttpStatus.CREATED)
    fun signUp(@RequestBody request: @Valid SignUpRequest): AccountDto {
        return accountService.createAccount(request)
    }

    @PostMapping("/signin")
    @ResponseStatus(HttpStatus.OK)
    fun signIn(@RequestBody request: @Valid SignInRequest): AuthDto {
        return accountService.accountValidation(request)
    }

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    fun authCheck(): String {
        return "OK"
    }
}